raiden/tests/utils/eth_node.py

```diff
@@ -426,6 +430,23 @@ def _validate_jsonrpc_result(result):
             stack.enter_context(executor)
             executors.append(executor)
 
+            # The timeout_limit_teardown is necessary to prevent the build
+            # being killed because of the lack of output, at the same time the
+            # timeout must never happen, because if does, not all finalizers
+            # are executed, leaving dirty state behind and result in test
+            # flakiness.
+            #
+            # Because of this, this value is arbitrarily smaller than the
+            # teardown timeout, forcing the subprocess to be killed on a timely
+            # manner, which should allow the teardown to proceed and finish
+            # before the timeout elapses.
+            timeout = 0.5
+
+            # The timeout values for the startup and teardown must be
+            # different, however the library doesn't support it. So here we
+            # must poke at the private member and overwrite it.
```

Comment:
ouch, I can see us getting bitten by this if `mirakuru` changes the implementation variables in the future. But well I guess tests will catch it.